scipy/linalg/decomp.py

```diff
@@ -1342,64 +1345,67 @@ def cdf2rdf(w, v):
            [ 0.     ,  2.59153,  3.23942],
            [ 0.     , -3.23942,  2.59153]])
     """
-    w1, v1 = _asarray_validated(w), _asarray_validated(v)
-    if w1.ndim < 1 or w1.ndim > 2:
-        raise ValueError('expected w to be a one-dimensional array or '
-                         'two-dimensional stacked array')
-    if w1.shape[-1] < 2:
-        raise ValueError('expected w array to have 2 or more elements')
-    if v1.ndim < 2 or v1.ndim > 3:
-        raise ValueError('expected v to be a two-dimensional array or '
-                         'three-dimensional stacked array')
-    if v1.shape[-2] != v1.shape[-1] or w1.shape[-1] < 2:
-        raise ValueError('expected v to be a square matrix or stacked square '
-                         'matrices: v1.shape[-2] = v1.shape[-1]')
-    if len(v1.shape) != len(w1.shape) + 1:
+    w, v = _asarray_validated(w), _asarray_validated(v)
+
+    # check dimensions
+    if w.ndim < 1:
+        raise ValueError('expected w to be at least one-dimensional')
+    if v.ndim < 2:
+        raise ValueError('expected v to be at least two-dimensional')
+    if v.ndim != w.ndim + 1:
         raise ValueError('expected eigenvectors array to have exactly one '
                          'dimension more than eigenvalues array')
-    if v1.shape[-1] != w1.shape[-1]:
+
+    # check shapes
+    n = w.shape[-1]
+    M = w.shape[:-1]
+    if v.shape[-2] != v.shape[-1]:
+        raise ValueError('expected v to be a square matrix or stacked square '
+                         'matrices: v.shape[-2] = v.shape[-1]')
+    if v.shape[-1] != n:
         raise ValueError('expected the same number of eigenvalues as '
                          'eigenvectors')
 
-    # number of stacked matrices and number of eigenvalues in each matrix
-    if w.ndim == 1:
-        stacked = False
-        M, n = 1, w.size
-        w = w[newaxis, :]
-        v = v[newaxis, :, :]
-    else:
-        stacked = True
-        M, n = w.shape
-
     # get indices for each first pair of complex eigenvalues
-    idx_im = argwhere(iscomplex(w))
+    complex_mask = iscomplex(w)
+    n_complex = complex_mask.sum(axis=-1)
 
     # check if all complex eigenvalues have conjugate pairs
-    if len(idx_im[::2, 0]) != len(idx_im[1::2, 0]):
+    if not (n_complex % 2 == 0).all():
         raise ValueError('expected complex-conjugate pairs of eigenvalues')
 
+    # find complex indices
+    idx = nonzero(complex_mask)
+    idx_stack = idx[:-1]
+    idx_elem = idx[-1]
+
+    # filter them to conjugate indices, assuming pairs are not interleaved
+    j = idx_elem[0::2]
+    k = idx_elem[1::2]
+    stack_ind = ()
+    for i in idx_stack:
+        # should never happen, assuming nonzero orders by the last axis
+        assert (i[0::2] == i[1::2]).all(), "Conjugate pair spanned different arrays!"
```

Comment:
Don't use `assert` for argument validation.  If the input is not as expected, raise a `ValueError`.  (But maybe you did this because this should never happen?)